spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/AdminApplication.java

```diff
@@ -18,13 +18,14 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.annotation.ComponentScan;
+import org.springframework.cloud.data.rest.config.AdminConfiguration;
+import org.springframework.context.annotation.Import;
 
 /**
  * @author Mark Fisher
  */
 @SpringBootApplication
-@ComponentScan(basePackages = "org.springframework.cloud.data.rest.controller")
```

Comment:
Can the `ComponentScan` help scanning more controllers in future? That way, we don't have to instantiate the `controller` beans inside AdminConfiguration. Instead we can `autowire` the module deployer into the controllers with component scan enabled in this class.